From e4fc38d63864809e3a6851c1a8d96f78761b6940 Mon Sep 17 00:00:00 2001
From: ming_qian <ming.qian@nxp.com>
Date: Thu, 21 Nov 2019 14:54:52 +0800
Subject: [PATCH 16/37] LF-80:[8QM_MEK/8QXP_MEK]mxc:vpu_malone:clear eos flag
 when output streamoff

commit 87fc389606f1e30ec0c4458316a206366007bdaa from
https://source.codeaurora.org/external/imx/linux-imx.git

if the eos flag is not clear after output streamoff,
it may cause firmware hang with multi instances

Signed-off-by: ming_qian <ming.qian@nxp.com>
(cherry picked from commit a1e9889cc10a99cadd004aa999fd7a098d4fa10b)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mxc/vpu_malone/vpu_b0.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/mxc/vpu_malone/vpu_b0.c b/drivers/mxc/vpu_malone/vpu_b0.c
index 88555d12cfd6..92b9973d62f5 100644
--- a/drivers/mxc/vpu_malone/vpu_b0.c
+++ b/drivers/mxc/vpu_malone/vpu_b0.c
@@ -1129,8 +1129,10 @@ static void clear_queue(struct queue_data *queue)
 	struct vb2_data_req *p_data_req = NULL;
 	struct vb2_data_req *p_temp;
 	struct vb2_buffer *vb;
+	struct vpu_ctx *ctx;
 
 	vpu_dbg(LVL_BIT_FUNC, "%s() is called\n", __func__);
+	ctx = container_of(queue, struct vpu_ctx, q_data[queue->type]);
 	check_queue_is_releasd(queue, "clear queue");
 
 	list_for_each_entry_safe(p_data_req, p_temp, &queue->drv_q, list) {
@@ -1143,6 +1145,8 @@ static void clear_queue(struct queue_data *queue)
 			vb2_buffer_done(vb, VB2_BUF_STATE_ERROR);
 	}
 	INIT_LIST_HEAD(&queue->drv_q);
+	if (queue->type == V4L2_SRC)
+		ctx->eos_stop_received = false;
 	vpu_dbg(LVL_BIT_FRAME_COUNT,
 		"%s qbuf_count : %ld, dqbuf_count : %ld\n",
 		queue->type == V4L2_DST ? "CAPTURE" : "OUTPUT",
-- 
2.17.1

